browser/ui/webui/ai_chat/ai_chat_untrusted_conversation_ui.cc

```diff
@@ -0,0 +1,208 @@
+    content::BrowserContext* browser_context) {
+  // Only enabled if we have a valid service
+  return (ai_chat::AIChatServiceFactory::GetForBrowserContext(
+              browser_context) != nullptr);
+}
+
+std::unique_ptr<content::WebUIController>
+AIChatUntrustedConversationUIConfig::CreateWebUIController(
+    content::WebUI* web_ui,
+    const GURL& url) {
+  return std::make_unique<AIChatUntrustedConversationUI>(web_ui);
+}
+
+AIChatUntrustedConversationUIConfig::AIChatUntrustedConversationUIConfig()
+    : WebUIConfig(content::kChromeUIUntrustedScheme,
+                  kAIChatUntrustedConversationUIHost) {}
+
+AIChatUntrustedConversationUIConfig::~AIChatUntrustedConversationUIConfig() =
+    default;
+
+AIChatUntrustedConversationUI::AIChatUntrustedConversationUI(
+    content::WebUI* web_ui)
+    : ui::MojoWebUIController(web_ui) {
+  // Create a URLDataSource and add resources.
+  content::WebUIDataSource* source = content::WebUIDataSource::CreateAndAdd(
+      web_ui->GetWebContents()->GetBrowserContext(),
+      kAIChatUntrustedConversationUIURL);
+  webui::SetupWebUIDataSource(source, kAiChatUiGenerated,
+                              IDR_AI_CHAT_UNTRUSTED_CONVERSATION_UI_HTML);
+
+  for (const auto& str : ai_chat::GetLocalizedStrings()) {
+    source->AddString(str.name,
+                      brave_l10n::GetLocalizedResourceUTF16String(str.id));
+  }
+
+  constexpr bool kIsMobile = BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_IOS);
+  source->AddBoolean("isMobile", kIsMobile);
+
+  source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::ScriptSrc,
+      "script-src 'self' chrome-untrusted://resources;");
+  source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::StyleSrc,
+      "style-src 'self' 'unsafe-inline' chrome-untrusted://resources;");
+  source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::ImgSrc,
+      "img-src 'self' blob: chrome-untrusted://resources;");
+  source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::FontSrc,
+      "font-src 'self' chrome-untrusted://resources;");
+  source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::FrameAncestors,
+      base::StringPrintf("frame-ancestors %s;", kAIChatUIURL));
+  source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::TrustedTypes, "trusted-types default;");
+}
+
+AIChatUntrustedConversationUI::~AIChatUntrustedConversationUI() = default;
+
+void AIChatUntrustedConversationUI::BindInterface(
+    mojo::PendingReceiver<ai_chat::mojom::UntrustedUIHandler> receiver) {
+  ui_handler_ = std::make_unique<UIHandler>(web_ui()->GetWebContents(),
+                                            std::move(receiver));
+}
+
+void AIChatUntrustedConversationUI::BindInterface(
+    mojo::PendingReceiver<ai_chat::mojom::UntrustedConversationHandler>
+        receiver) {
+  // Get conversation from URL
```

Comment:
please add comments/docs explaining the url format somewhere

Reply:
Added `brave/browser/ai_chat/ai_chat_urls.h`:
```cpp
namespace ai_chat {

// UI that will open a conversation associated with the active Tab in the same
// browser window. The conversation will change when that Tab navigates.
GURL TabAssociatedConversationUrl();

// UI that will open to a specific conversation. The conversation will not
// change upon any navigation.
GURL ConversationUrl(std::string_view conversation_uuid);

// Extracts the conversation UUID from a conversation URL or a conversation
// entries iframe
std::string_view ConversationUUIDFromURL(const GURL& url);

}  // namespace ai_chat
```